DataStructures/BitList.cs

```diff
@@ -0,0 +1,1750 @@
+            }
+            catch
+            {
+                throw new FileWritingException();
+            }
+        }
+
+        /// <summary>
+        /// Convert this instance to a text rapresentation.
+        /// </summary>
+        public string ToText()
+        {
+            if (IsEmpty)
+            {
+                return string.Empty;
+            }
+
+            var text = string.Empty;
+            foreach (var bit in ToBooleanArray())
+            {
+                text += bit ? "1" : "0";
+            }
+
+            return text;
+        }
+
+        /// <summary>
+        /// Convert this instance to a specified unmanaged non primitive value array.
+        /// </summary>
+        /// <typeparam name="T"></typeparam>
+        /// <returns></returns>
+        public T[] ToUnmanagedNonPrimitiveValueArray<T>() where T : unmanaged
+        {
+            try
+            {
+                if (typeof(T).IsPrimitive || typeof(T).IsPointer)
+                {
+                    throw new ConversionErrorException();
+                }
+                else if (IsEmpty)
+                {
+                    return new T[0];
+                }
+                else
+                {
+                    var bytes = ToByteArray().ToList();
+
+                    //Unit of measure: byte (8 bit).
+                    var size = Marshal.SizeOf(typeof(T).IsEnum ? Enum.GetUnderlyingType(typeof(T)) : typeof(T));
+
+                    while (bytes.Count % size != 0)
+                    {
+                        //Add a missing byte.
+                        bytes.Add(0);
+                    }
+
+                    var result = new List<T>(); //A list of unmanaged non primitive value type objects.
+                    for (var i = 0; i < bytes.Count; i += size)
+                    {
+                        if (typeof(T).IsEnum)
+                        {
+                            T obj;
+                            switch (Enum.GetUnderlyingType(typeof(T)).Name)
+                            {
+                                case "Byte":
+                                    obj = (T)Enum.ToObject(typeof(T), bytes[i]);
+                                    break;
+
+                                case "SByte":
+                                    obj = (T)Enum.ToObject(typeof(T), (sbyte)bytes[i]);
+                                    break;
+
+                                case "Int16":
+                                    obj = (T)Enum.ToObject(typeof(T), BitConverter.ToInt16(bytes.GetRange(i, 2).ToArray()));
+                                    break;
+
+                                case "UInt16":
+                                    obj = (T)Enum.ToObject(typeof(T), BitConverter.ToUInt16(bytes.GetRange(i, 2).ToArray()));
+                                    break;
+
+                                default:
+                                case "Int32":
```

Comment:
Codacy found an issue: [Remove this empty 'case' clause.](https://app.codacy.com/manual/siriak/C-Sharp/pullRequest?prid=5559704)